module/Finna/src/Finna/Controller/MyResearchController.php

```diff
@@ -1043,6 +1043,80 @@ public function importAction()
     {
     }
 
+    /**
+     * Display dialog for importing public favorites lists across organisations.
+     *
+     * @return mixed
+     */
+    public function importPublicListAction()
+    {
+        if (!($user = $this->getUser())) {
+            return $this->forceLogin();
+        }
+        $view = $this->createViewModel();
+        $view->lists = $user->getLists();
+        $view->url = "";
+        if ($this->formWasSubmitted('import_public_list_form')) {
+            $publicUserListUrl = $this->params()->fromPost('url');
+            $view->url = $publicUserListUrl;
+            $targetListId = $this->params()->fromPost('list');
+            $matches = [];
+            $urlValidator = "/^([a-z]+):\/\/((?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9])\/(\w+\/)?List\/(.+)$/";
+            preg_match(
+                $urlValidator,
+                $publicUserListUrl,
+                $matches
+            );
+            if (empty($matches)) {
+                $this
```

Comment:
Vähemmän rivinvaihtoja,  $this->flashMessenger() menee hyvin yhdelle riville. Sama muille monirivisille alempana.